app/code/Magento/Sales/Block/Order/Email/Items/Order/DefaultOrder.php

```diff
@@ -93,10 +96,10 @@ public function getProductAdditionalInformationBlock()
     /**
      * Get the html for item price
      *
-     * @param OrderItem $item
+     * @param OrderItem|InvoiceItem|CreditmemoItem  $item
      * @return string
      */
-    public function getItemPrice(OrderItem $item)
+    public function getItemPrice(AbstractModel $item)
```

Comment:
The problem is: in case somebody extended the class and had this function overriden, the code will result in the following warning:
PHP Warning:  Declaration of ChildClass:: getItemPrice(OrderItem $item) should be compatible with DefaultOrder:: getItemPrice(AbstractModel $item).

To avoid breaking backward compatibility please create a separate implementation (class and/or method) for this method(s) that will accept different types of items or simply abstract item. Then use the new implementation in the client code.